contracts/helpers/TransferHelper.sol

```diff
@@ -70,97 +88,242 @@ contract TransferHelper is TransferHelperInterface, TokenTransferrer {
                         item.token,
                         msg.sender,
                         recipient,
                         item.identifier,
                         item.amount
                     );
+                } else {
+                    // Revert if the item being transferred is a native token.
+                    revert InvalidItemType();
                 }
             }
+        }
+    }
+
+    /**
+     * @notice Perform multiple transfers to a single recipient via
+     *         the conduit derived from the provided conduit key.
+     *
+     * @param items      The items to transfer.
+     * @param recipient  The address the items should be transferred to.
+     * @param conduitKey The conduit key referring to the conduit through
+     *                   which the bulk transfer should occur.
+     */
+    function _performTransfersWithConduit(
+        TransferHelperItem[] calldata items,
+        address recipient,
+        bytes32 conduitKey
+    ) internal {
+        // Retrieve total number of transfers and place on stack.
+        uint256 totalTransfers = items.length;
+
+        // Derive the conduit address from the deployer, conduit key
+        // and creation code hash.
+        address conduit = address(
+            uint160(
+                uint256(
+                    keccak256(
+                        abi.encodePacked(
+                            bytes1(0xff),
+                            address(_CONDUIT_CONTROLLER),
+                            conduitKey,
+                            _CONDUIT_CREATION_CODE_HASH
+                        )
+                    )
+                )
+            )
+        );
 
-            // Call the conduit and execute bulk transfers.
-            ConduitInterface(conduit).execute(conduitTransfers);
+        // Declare a new array to populate with each token transfer.
+        ConduitTransfer[] memory conduitTransfers = new ConduitTransfer[](
+            totalTransfers
+        );
+
+        // Skip overflow checks: all for loops are indexed starting at zero.
+        unchecked {
+            // Iterate over each transfer.
+            for (uint256 i = 0; i < totalTransfers; ++i) {
+                // Retrieve the transfer in question.
+                TransferHelperItem calldata item = items[i];
+
+                // Create a ConduitTransfer corresponding to each
+                // TransferHelperItem.
+                conduitTransfers[i] = ConduitTransfer(
+                    item.itemType,
+                    item.token,
+                    msg.sender,
+                    recipient,
+                    item.identifier,
+                    item.amount
+                );
+            }
         }
 
-        // Return a magic value indicating that the transfers were performed.
-        magicValue = this.bulkTransfer.selector;
+        // Check if the derived conduit is a contract.
+        if (!_isContract(conduit)) {
```

Comment:
This check is unnecessary as a magic value must be returned by the call (and won’t be if there’s no contract) — would lose the InvalidMagicValue custom error and use InvalidConduit in its place.